spack/package.py

```diff
@@ -93,18 +95,20 @@ class Arbor(CMakePackage, CudaPackage):
     depends_on("py-mpi4py", when="+mpi+python", type=("build", "run"))
 
     # python (bindings)
-    extends("python", when="+python")
-    depends_on("python@3.7:", when="+python", type=("build", "run"))
-    depends_on("py-numpy", when="+python", type=("build", "run"))
     with when("+python"):
+        extends("python")
+        depends_on("python@3.7:", type=("build", "run"))
+        depends_on("python@3.8:", when="@0.9.1:", type=("build", "run"))
+        depends_on("py-numpy", type=("build", "run"))
         depends_on("py-pybind11@2.6:", type="build")
```

Comment:
Uh. This version is all over the place,
- 0.9.1: 2.6 which is quite old
- 0.7.1: 2.10 
- 0.5.3: 2.8

or am I misunderstand something.

Reply:
Ok, that makes it way less convoluted!